app/components/App.jsx

```diff
@@ -130,21 +134,41 @@ class App extends Component {
     });
   }
 
+  toggleHamburgerMenu() {
+    this.setState((state, props) => ({hamburgerMenuIsOpen: !state.hamburgerMenuIsOpen}));
+  }
+
+  onHamburgerMenuStateChange(state) {
+    this.setState({hamburgerMenuIsOpen: state.isOpen});
```

Comment:
is there a reason you set the state differently here than above in the toggleHamburgerMenu function? If not, I think it would be better to choose one way and be consistent.

Reply:
This is the API that the [`react-burger-menu`](https://github.com/negomi/react-burger-menu) exposes. It expects you to pass in a callback where the first argument is the new state. The API is honestly a bit confusing since it looks like you can use the component in two ways:

1. react-burger-menu maintains its own state and you don't pass in the `isOpen` prop into the component
2. react-burger-menu maintains its own state *and* you can pass in the `isOpen` prop into the component, meaning that you must synchronize state both inside and outside of the component

I opted for 2) because based on Derek's designs, we needed a lot more control over the actual hamburger button than the library provides by itself. If you can think of a better way of handling this, I'm more than happy to change it.

Reply:
Sounds good to me